cads_adaptors/tools/url_tools.py

```diff
@@ -35,21 +36,20 @@ def try_download(urls: List[str]) -> List[str]:
     for url in urls:
         path = urllib.parse.urlparse(url).path.lstrip("/")
         dir = os.path.dirname(path)
-        os.makedirs(dir, exist_ok=True)
+        if dir:
+            os.makedirs(dir, exist_ok=True)
         try:
             multiurl.download(url, path)
+        except Exception as exc:
+            excs.append({url: exc})
+            logger.warning(f"Failed download for URL: {url}\nTraceback: {exc}")
+        else:
             paths.append(path)
-        except requests.exceptions.HTTPError as exc:
-            if exc.response is not None and exc.response.status_code == 404:
-                logger.warning(exc)
-                excs.append(exc)
-            else:
-                raise exc
+
     if len(paths) == 0:
         raise RuntimeError(
-            f"Request empty. At least one of the following {urls} "
-            "must be a valid url from which to download the data "
-            f"download errors: {[str(exc) for exc in excs]}"
+            f"Request empty. At least one of the following:\n{yaml.safe_dump(urls, indent=2)} "
```

Comment:
Two ideas:
1. Avoid using an external package: `"  "+"\n  ".join(urls)`
2. If you want to restore the display of all exceptions, python3.11 introduces ExceptionGroup (we are already testing py3.11, we will eventually move from 3.10 to 3.11). Here is an example from cacholote: https://github.com/ecmwf-projects/cacholote/blob/bf87b56a7f160de61ab644932b50ad66515accc3/cacholote/encode.py#L164-L166

Reply:
I think error messaging layout can be addressed in a separate ticket, let's just get this merged so it's working